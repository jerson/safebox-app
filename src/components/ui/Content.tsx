import React from 'react';
import {
  StyleProp,
  StyleSheet,
  View,
  ViewProps,
  ViewStyle
} from 'react-native';

const styles = StyleSheet.create({
  content: {
    paddingHorizontal: 20
  } as ViewStyle,
  center: {
    justifyContent: 'center'
  } as ViewStyle,
  vertical: {
    flex: 1,
    paddingTop: 20
  } as ViewStyle
});

export interface ContentProps extends ViewProps {
  style?: StyleProp<ViewStyle>;
  /**
   * Usa `center` para centrar el contenido con `justifyContent`
   * @default false
   * @
   */
  center?: boolean;
  /**
   * Usa `vertical` para agregar `flex` y un `paddingTop`
   * @default true
   */
  vertical?: boolean;
  children: React.ReactNode | React.ReactNode[];
}

function Content(props: ContentProps) {
  const { style, center, vertical, children, ...extraProps } = props;
  return (
    <View
      style={[
        styles.content,
        center && styles.center,
        vertical && styles.vertical,
        style
      ]}
      {...extraProps}
    >
      {children}
    </View>
  );
}

Content.defaultProps = {
  center: false,
  vertical: true
};

export default Content;
