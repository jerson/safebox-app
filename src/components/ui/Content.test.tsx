import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Content, { ContentProps } from './Content';

const props: ContentProps = {
  style: {}
};

it('renders without crashing', () => {
  const snapshot = renderer.create(<Content {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

it('render just views', () => {
  const snapshot = renderer.create(<Content {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
