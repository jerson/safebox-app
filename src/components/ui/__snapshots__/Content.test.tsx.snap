// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
      },
      false,
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {},
    ]
  }
/>
`;
